mmaction/models/localizers/ssn.py

```diff
@@ -0,0 +1,122 @@
+        self.init_weights()
+
+    def init_weights(self):
+        super(SSN, self).init_weights()
+        self.backbone.init_weights()
+        self.cls_head.init_weights()
+
+    def extract_feat(self, imgs):
+        """Extract features through a backbone.
+
+        Args:
+            imgs (torch.Tensor): The input images.
+        Returns:
+            torch.tensor: The extracted features.
+        """
+        x = self.backbone(imgs)
+        return x
+
+    def forward_train(self, imgs, proposal_scale_factor, proposal_type,
+                      proposal_labels, reg_targets, **kwargs):
+        imgs = imgs.reshape((-1, self.in_channels) + imgs.shape[4:])
+
+        x = self.extract_feat(imgs)
+
+        if self.avg_pool:
+            x = self.avg_pool(x)
+        if self.dropout is not None:
+            x = self.dropout(x)
+
+        activity_score, completeness_score, bbox_pred = self.cls_head(
+            (x, proposal_scale_factor))
+
+        loss = self.loss_cls(activity_score, completeness_score, bbox_pred,
+                             proposal_type, proposal_labels, reg_targets,
+                             self.train_cfg)
+        loss_dict = dict(**loss)
+
+        return loss_dict
+
+    def forward_test(self, imgs, relative_proposal_list, scale_factor_list,
+                     proposal_tick_list, reg_stats, **kwargs):
+        num_crop = imgs.shape[0]
+        imgs = imgs.reshape((num_crop, -1, self.in_channels) + imgs.shape[3:])
+        num_ticks = imgs.shape[1]
+
+        output = []
+        minibatch_size = self.test_cfg.ssn.sampler.batch_size
+        for ind in range(0, num_ticks, minibatch_size):
+            chunk = imgs[:, ind:ind + minibatch_size,
+                         ...].view((-1, ) + imgs.shape[2:])
+            x = self.extract_feat(chunk.cuda())
+            if self.avg_pool:
+                x = self.avg_pool(x)
+            # Merge crop to save memory.
+            x = x.reshape((num_crop, x.size(0) // num_crop, -1)).mean(dim=0)
+            output.append(x)
+        output = torch.cat(output, dim=0)
+
+        relative_proposal_list = relative_proposal_list.squeeze(0)
+        proposal_tick_list = proposal_tick_list.squeeze(0)
+        scale_factor_list = scale_factor_list.squeeze(0)
+        reg_stats = reg_stats.squeeze(0)
+
+        if not self.is_test_prepared:
+            self.is_test_prepared = self.cls_head.prepare_test_fc(
+                self.cls_head.consensus.feat_multiplier)
+
+        (output, activity_scores, completeness_scores,
+         bbox_preds) = self.cls_head(
+             (output, proposal_tick_list, scale_factor_list), test_mode=True)
+
+        if bbox_preds is not None:
+            bbox_preds = bbox_preds.view(-1, self.cls_head.num_classes, 2)
+            bbox_preds[:, :, 0] = (
+                bbox_preds[:, :, 0] * reg_stats[1, 0] + reg_stats[0, 0])
+            bbox_preds[:, :, 1] = (
+                bbox_preds[:, :, 1] * reg_stats[1, 1] + reg_stats[0, 1])
```

Comment:
what does `stats` mean?

Reply:
Maybe statistics? https://github.com/open-mmlab/mmaction2/pull/37/files#diff-18d0155e53255529a7a5cede43a65651R143

Reply:
you can rename it now